.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs

```diff
@@ -732,42 +732,27 @@ private string ParseTimePeriod(Match match, out int beginHour, out int endHour)
             }
 
             // Parse "pm"
-            string matchPmStr = string.Empty, matchAmStr = string.Empty, descStr = string.Empty;
-            string matchPmStr1 = string.Empty, matchAmStr1 = string.Empty, descStr1 = string.Empty;
-            var matchPmStrGroup = match.Groups[Constants.PmGroupName];
-            var matchAmStrGroup = match.Groups[Constants.AmGroupName];
+            string descStr = string.Empty, beginDescStr = string.Empty, endDescStr = string.Empty;
+            var matchPmStr = match.Groups[Constants.PmGroupName].Value;
+            var matchAmStr = match.Groups[Constants.AmGroupName].Value;
             var descStrGroup = match.Groups[Constants.DescGroupName];
-            if (matchPmStrGroup.Captures.Count > 0)
-            {
-                matchPmStr = matchPmStrGroup.Captures[0].Value;
-                if (matchPmStrGroup.Captures.Count > 1)
-                {
-                    matchPmStr1 = matchPmStrGroup.Captures[1].Value;
-                }
-            }
 
-            if (matchAmStrGroup.Captures.Count > 0)
+            if (descStrGroup.Captures.Count > 1)
             {
-                matchAmStr = matchAmStrGroup.Captures[0].Value;
-                if (matchAmStrGroup.Captures.Count > 1)
-                {
-                    matchAmStr1 = matchAmStrGroup.Captures[1].Value;
-                }
-            }
+                // first particle am/pm in patterns like "between 8am and 2pm"
+                beginDescStr = descStrGroup.Captures[0].Value;
```

Comment:
This is still using capture index groups. Why not just use the new names in the modified regex?